lib/process-services-cloud/src/lib/form/components/form-cloud.component.spec.ts

```diff
@@ -237,6 +237,48 @@ describe('FormCloudComponent', () => {
         expect(formComponent.getFormByTaskId).toHaveBeenCalledWith(appName, taskId);
     });
 
+    it('should call the process storage to retrieve the folder with only the taskId', fakeAsync(() => {
+        spyOn(formComponent, 'getFormByTaskId').and.stub();
+        spyOn(formCloudService, 'getProcessStorageFolderTask')
+            .and.returnValue( of({nodeId : '123', path: '/a/path/type', type: 'fakeType'}));
+        const taskId = '<task id>';
+        const appName = 'test-app';
+        formComponent.appName = appName;
+        formComponent.form = new FormCloud({id : 'fake-form-id'});
+        formComponent.form['hasUpload'] = true;
+        formComponent.taskId = taskId;
+        formComponent.readOnly = false;
+
+        const change = new SimpleChange(null, 'new-app-name', true);
+        formComponent.ngOnChanges({ 'appName': change });
+        tick();
+        expect(formCloudService.getProcessStorageFolderTask).toHaveBeenCalledWith(appName, taskId, undefined);
+        expect(formComponent.form.nodeId).toBe('123');
+        expect(formComponent.form.contentHost).toBe('/a/path/type');
+    }));
+
+    it('should call the process storage to retrieve the folder with taskId and processInstanceId', fakeAsync(() => {
+        spyOn(formComponent, 'getFormByTaskId').and.stub();
+        spyOn(formCloudService, 'getProcessStorageFolderTask')
+            .and.returnValue( of({nodeId : '123', path: '/a/path/type', type: 'fakeType'}));
+        const taskId = '<task id>';
+        const processInstanceId = 'i-am-the-process-instance-id';
+        const appName = 'test-app';
+        formComponent.appName = appName;
+        formComponent.form = new FormCloud({id : 'fake-form-id'});
+        formComponent.form['hasUpload'] = true;
+        formComponent.taskId = taskId;
+        formComponent.processInstanceId = processInstanceId;
+        formComponent.readOnly = false;
+
+        const change = new SimpleChange(null, 'new-app-name', true);
+        formComponent.ngOnChanges({ 'appName': change });
```

Comment:
the ng change should have taskId and processId as input instead of override it from the parent
```
   formComponent.taskId = taskId;
   formComponent.processInstanceId = processInstanceId;```
otherwise we will not test this code
```
  ngOnChanges(changes: SimpleChanges) {
        const appName = changes['appName'];
        if (appName && appName.currentValue) {
            if (this.taskId && this.processInstanceId) {
                this.getFormDefinitionWithFolderTask(this.appName, this.taskId, this.processInstanceId);
            } else if (this.taskId) {
                this.getFormByTaskId(this.appName, this.taskId);
            }
```

Reply:
That condition is changed, it's enough the taskid